test.sh

```diff
@@ -16,4 +16,9 @@
 
 # prow calls this file currently, but we can just use `make test` to test
 # the set of things we want.
-make test
+CHECK_DOCS_ONLY=$("test/check-docs-only.sh")
+if [ -z "$CHECK_DOCS_ONLY" ]; then
+    make test
+    exit 0
+fi
```

Comment:
The motivation for this task is: 
- If we just change the docs, we do not need to run all e2e tests for example
- So that we do not need wait long to check the results or spend an unnecessary resource to do that 

However, see that with your test with the PR: 

a) I could not find the WARNING in for example : https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kubebuilder/2688/pull-kubebuilder-e2e-k8s-1-23-3/1525693478816714752/build-log.txt. So are we really skipping the tests? 
b) we still checking that the kind and everything were start-ups and it still consuming a time to be executed; 

<img width="864" alt="Screenshot 2022-05-18 at 10 39 49" src="https://user-images.githubusercontent.com/7708031/169009375-b749a4d5-2610-4442-8608-4e91ce23fc8c.png">

So, we did not achieve the goals/motivations so far. Also, by looking at https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/kubebuilder/kubebuilder-presubmits.yaml#L5 I begin to wonder:

If we skip here would we still be on the prow for starting a cluster? If yes, then how much do we actually win doing the change in this way? We might realise that unless we are able to skip there we do not really add value in adding this one here.

Reply:
See: https://stackoverflow.com/questions/23303549/what-are-commit-ish-and-tree-ish-in-git
One way to move forward here, I think is:
- create a Makefile target that will check the docs 
- the check its result here to use in the e2e 
So that we also let the check available to contributors. 

c/c @NikhilSharmaWe